### DISCUSSION TITLE
Input redirection

### DISCUSSION THREAD

**Discussion by ineu** [nushell/nushell#11587] [Ideas]
So we have `o>`, `e>`, `o+e>`, `|` - all for the data flowing *out* of a command. But there's nothing for the data flowing *into* the command, i.e. no input redirection. Which it's often handy: if I need to pipe a file into a command, the only way (I'm aware of) to do it in Nu is `cat /dev/zero | pv -S -s 1g o> file`. In bash it could be shorter: `pv -S -s 1g < /dev/zero > file`.

So are we stuck with invoking extra `cat`/`open`, or is it something that can be added to the shell in future? It's not a feature request, but rather something to discuss.

**Comment by fdncred** [comment]
Correct, we don't support stdin redirection right now. I'm not sure if we'll support it in the future. I would be good to discuss it. Thanks for bringing this up.

**Reply by emailstorbala** [reply]
This is really a critical feature @fdncred . It would be a handicap without that feature. I was mentioning this only in the other thread - https://github.com/nushell/nushell/discussions/11680. Without this feature, it would be difficult to automate certain functionality which involves passing stdin to a program like confirmation, other runtime inputs excetra. Kindly reconsider this feature.

**Reply by fdncred** [reply]
You should feel free to submit a PR with this feature.
